vermouth/processors/do_links.py

```diff
@@ -33,16 +33,24 @@ def _atoms_match(node1, node2):
     mods = []
     for mod in node1.get('modifications', []):
         mods.extend(mod.name)
-    # Case 1: link does not restrict modifications: always match
-    # Case 2: link specifies modifications, but molecule node has none: don't match
-    # Case 3: both have modifications: match iff all molecule mods match link mods
-    # For case 3 we need to do a little jiggery-pokery to leverage attributes_match.
-    # This probably means that that functions need to be cut up into smaller
-    # pieces.
-    mods_match = ('modifications' not in node2 or  # Case 1
-                  (mods and  # Case 2
-                   all(attributes_match({'_': modname}, {'_': node2['modifications']}) for modname in mods)))  # Case 3
-    return mods_match and attributes_match(node1, node2, ignore_keys=('order', 'replace', 'modifications'))
+
+                  # No modifications specified by link: always match
+    mods_match = ('modifications' not in node2 or
+                  # empty modifications in link and no modifications in molecule: match
+                  (not node2['modifications'] and not mods) or
+                  # Else, if both specify modifications, then...
+                  (node2['modifications'] and mods and
+                   # link modifications must be a list, and molecule mods must
+                   # match links mods exactly
+                   ((isinstance(node2['modifications'], list) and sorted(mods) == sorted(node2['modifications'])) or
+                    # Or link mods are a simple string or a Choice, and all
+                    # molecule modifications must be accounted for
+                    # Here we need to do a little jiggery-pokery to leverage
+                    # attributes_match. This probably means that that function
+                    # needs to be cut up into smaller pieces.
+                    all(attributes_match({'_': modname}, {'_': node2['modifications']}) for modname in mods))))
```

Comment:
this is rather horrible to read as code but for now is fine

Reply:
Yes, yes it is. I agree that it's fine for now.